src/certificate/mod.rs

```diff
@@ -0,0 +1,192 @@
+use std::{str::FromStr, time::Duration};
+
+use pkcs8::{der::EncodePem, spki::SubjectPublicKeyInfoOwned, EncodePublicKey, LineEnding};
+use x509_cert::{
+    builder::{Builder, CertificateBuilder, Profile},
+    certificate::CertificateInner,
+    name::Name,
+    serial_number::SerialNumber,
+    time::Validity,
+};
+
+use crate::{CryptoCoreError, Ed25519Keypair, Ed25519PublicKey, Ed25519Signature};
+
+fn default_validity_duration() -> Result<Validity, CryptoCoreError> {
+    // Approximate the number of seconds in a month (30.44 days)
+    const SECONDS_IN_MONTH: u64 = 30 * 24 * 60 * 60 + 15 * 60 * 60 + 26 * 60;
+
+    // Number of months
+    const MONTHS: u32 = 3;
+
+    // Calculate the total duration in seconds
+    let total_seconds = u64::from(MONTHS) * SECONDS_IN_MONTH;
```

Comment:
You could make this a `const` by changing the type of `MONTH` to `u64`.

Reply:
indeed, thks